src/main/groovy/life/qbic/registration/MeasurementID.groovy

```diff
@@ -0,0 +1,125 @@
+package life.qbic.registration
+
+import groovy.transform.EqualsAndHashCode
+import life.qbic.registration.types.QDatasetType
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+
+/**
+ * A measurement ID references an openBIS object code used for measurements
+ *
+ * @since 1.0.0
+ */
+@EqualsAndHashCode
+class MeasurementID {
```

Comment:
I think we can drop this class entirely, since the ID has already been processed and validated. The ETL should be as "stupid" as possible, just reading and executing the registration. Everything else is part of the pre-processing checks and part of the scanner application.